multimodal/src/autogluon/multimodal/learners/default_learner.py

```diff
@@ -0,0 +1,1724 @@
+        )
+        logits = extract_from_output(ret_type=ret_type, outputs=outputs)
+
+        # 6. get y_pred and y_true for computing metrics. TODO: refactor this into sub-learners
+        # i.e. define self.transform_logits_to_metric_input(logits)
+        metric_data = {}
+        if self._problem_type in [BINARY, MULTICLASS]:
+            y_pred_prob = logits_to_prob(logits)
+            metric_data[Y_PRED_PROB] = y_pred_prob
+
+        y_pred = self._df_preprocessor.transform_prediction(
+            y_pred=logits,
+            inverse_categorical=False,
+        )
+        y_pred_inv = self._df_preprocessor.transform_prediction(
+            y_pred=logits,
+            inverse_categorical=True,
+        )
+
+        # TODO: refactore this into NER learner
+        # i.e. define self.get_y_true_for_metric(data)
+        # if self._problem_type == NER:
+        #     y_true = self._df_preprocessor.transform_label_for_metric(df=data, tokenizer=self._model.tokenizer)
+        # else:
+        #     y_true = self._df_preprocessor.transform_label_for_metric(df=data)
+        y_true = self._df_preprocessor.transform_label_for_metric(df=data)
+
+        metric_data.update(
+            {
+                Y_PRED: y_pred,
+                Y_TRUE: y_true,
+            }
+        )
+
+        metrics_is_none = False
+
+        if metrics is None:
+            metrics_is_none = True
+            metrics = [self._eval_metric_name]
+        if isinstance(metrics, str):
+            metrics = [metrics]
+
+        # 7. compute metrics. TODO: Refactor this into sub-learners
+        # i.e. define self.compute_score_and_get_results()
+        results = {}
+        # if self._problem_type == NER:
+        #     score = compute_score(
+        #         metric_data=metric_data,
+        #         metric_name=self._eval_metric_name.lower(),
+        #     )
+        #     score = {k.lower(): v for k, v in score.items()}
+        #     if metrics_is_none:
+        #         results = score
+        #     else:
+        #         for per_metric in metrics:
+        #             if per_metric.lower() in score:
+        #                 results.update({per_metric: score[per_metric.lower()]})
+        #             else:
+        #                 logger.warning(f"Warning: {per_metric} is not a supported evaluation metric!")
+        #         if not results:
+        #             results = score  # If the results dict is empty, return all scores.
+        # else:
+        #     for per_metric in metrics:
+        #         score = compute_score(
+        #             metric_data=metric_data,
+        #             metric_name=per_metric.lower(),
+        #         )
+        #         results[per_metric] = score
+        for per_metric in metrics:
+            score = compute_score(
+                metric_data=metric_data,
+                metric_name=per_metric.lower(),
+            )
+            results[per_metric] = score
+
+        if return_pred:
```

Comment:
Why are we returning pred from an evaluate call??

Reply:
This is backwards. You should instead be able to pass pred as input to evaluate, rather than returning pred as output to evaluate.